File: services/education/internal/model/product_package.go
package model

import "time"

type ProductPackageQueryDTO struct {
	PageRequestModel PageRequestModel          `json:"pageRequestModel"`
	QueryModel       ProductPackageQueryFilter `json:"queryModel"`
	SortModel        map[string]any            `json:"sortModel"`
}

type ProductPackageQueryFilter struct {
	Name                     string                      `json:"name"`
	SearchKey                string                      `json:"searchKey"`
	OnlineSale               *bool                       `json:"onlineSale"`
	IsOnlineSaleMicoSchool   *bool                       `json:"isOnlineSaleMicoSchool"`
	IsShowMicoSchool         *bool                       `json:"isShowMicoSchool"`
	ProductPackageProperties []ProductPackagePropertyRef `json:"productPackageProperties"`
}

type ProductPackagePropertyRef struct {
	ProductPackagePropertyID    string `json:"productPackagePropertyId"`
	ProductPackagePropertyValue string `json:"productPackagePropertyValue"`
}

type ProductPackageItemMutation struct {
	ProductID      string  `json:"productId"`
	SkuID          string  `json:"skuId"`
	SkuCount       float64 `json:"skuCount"`
	FreeQuantity   float64 `json:"freeQuantity"`
	DiscountType   *int    `json:"discountType,omitempty"`
	DiscountNumber float64 `json:"discountNumber"`
}

type ProductPackageMutation struct {
	ID                       string                       `json:"id,omitempty"`
	Name                     string                       `json:"name"`
	OnlineSale               bool                         `json:"onlineSale"`
	IsAllowEditWhenEnroll    bool                         `json:"isAllowEditWhenEnroll"`
	Title                    string                       `json:"title"`
	Images                   string                       `json:"images"`
	Description              string                       `json:"description"`
	IsShowMicoSchool         bool                         `json:"isShowMicoSchool"`
	IsOnlineSaleMicoSchool   bool                         `json:"isOnlineSaleMicoSchool"`
	BuyRule                  map[string]any               `json:"buyRule"`
	Items                    []ProductPackageItemMutation `json:"items"`
	SubjectIDs               []int64                      `json:"subjectIds"`
	ProductPackageProperties []ProductPackagePropertyRef  `json:"productPackageProperties"`
}

type ProductPackageOperateMutation struct {
	ID                      string `json:"id"`
	OnlineSale              *bool  `json:"onlineSale,omitempty"`
	IsShowMicoSchool        *bool  `json:"isShowMicoSchool,omitempty"`
	IsOnlineSaleMicoSchool  *bool  `json:"isOnlineSaleMicoSchool,omitempty"`
	IsAllowEditWhenEnroll   *bool  `json:"isAllowEditWhenEnroll,omitempty"`
}

type ProductPackagePropertyVO struct {
	ProductPackagePropertyID        string `json:"productPackagePropertyId"`
	ProductPackagePropertyName      string `json:"productPackagePropertyName,omitempty"`
	ProductPackagePropertyValue     string `json:"productPackagePropertyValue"`
	ProductPackagePropertyValueName string `json:"productPackagePropertyValueName,omitempty"`
}

type ProductPackageItemVO struct {
	ID             string `json:"id"`
	ProductType    int    `json:"productType"`
	ProductID      string `json:"productId"`
	ProductName    string `json:"productName"`
	SkuID          string `json:"skuId"`
	SkuName        string `json:"skuName"`
	LessonScope    int    `json:"lessonScope"`
	LessonType     int    `json:"lessonType"`
	LessonMode     int    `json:"lessonMode"`
	LessonAudition bool   `json:"lessonAudition"`
}

type ProductPackageSubjectVO struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

type ProductPackageVO struct {
	ID                      string                     `json:"id"`
	Name                    string                     `json:"name"`
	Title                   string                     `json:"title"`
	OnlineSale              bool                       `json:"onlineSale"`
	IsOnlineSaleMicoSchool  bool                       `json:"isOnlineSaleMicoSchool"`
	IsShowMicoSchool        bool                       `json:"isShowMicoSchool"`
	OrgProductPackageID     string                     `json:"orgProductPackageId"`
	Editable                bool                       `json:"editable"`
	IsSyncOrgProductPackage bool                       `json:"isSyncOrgProductPackage"`
	Sale                    int                        `json:"sale"`
	TotalAmount             float64                    `json:"totalAmount"`
	DiscountAmount          float64                    `json:"discountAmount"`
	FinalAmount             float64                    `json:"finalAmount"`
	Images                  string                     `json:"images"`
	Subjects                []ProductPackageSubjectVO  `json:"subjects"`
	ExtendProperties        []ProductPackagePropertyVO `json:"extendProperties"`
	UpdatedTime             *time.Time                 `json:"updatedTime,omitempty"`
	Items                   []ProductPackageItemVO     `json:"items"`
}

type ProductPackagePagedResult struct {
	List  []ProductPackageVO `json:"list"`
	Total int                `json:"total"`
}

type ProductPackageStatistics struct {
	TotalCount  int `json:"totalCount"`
	OnSaleCount int `json:"onSaleCount"`
}

File: 5.0-admin/src/api/edu-center/product-package.ts
import { usePost } from '~/utils/request'

export interface ProductPackagePropertyRef {
  productPackagePropertyId: string
  productPackagePropertyValue: string
}

export interface ProductPackageItemMutation {
  productId: string
  skuId: string
  skuCount: number
  freeQuantity: number
  discountType?: number
  discountNumber: number
}

export interface ProductPackageItem {
  id: string
  productType: number
  productId: string
  productName: string
  skuId: string
  skuName: string
  lessonScope: number
  lessonType: number
  lessonMode: number
  lessonAudition: boolean
}

export interface ProductPackageProperty {
  productPackagePropertyId: string
  productPackagePropertyName?: string
  productPackagePropertyValue: string
  productPackagePropertyValueName?: string
}

export interface ProductPackageInfo {
  id: string
  name: string
  title: string
  onlineSale: boolean
  isOnlineSaleMicoSchool: boolean
  isShowMicoSchool: boolean
  orgProductPackageId: string
  editable: boolean
  isSyncOrgProductPackage: boolean
  sale: number
  totalAmount: number
  discountAmount: number
  finalAmount: number
  images: string
  subjects: Array<{ id: string, name: string }>
  extendProperties: ProductPackageProperty[]
  updatedTime?: string
  items: ProductPackageItem[]
}

export interface ProductPackagePagedResult {
  list?: ProductPackageInfo[]
  total?: number
}

export interface ProductPackageStatistics {
  totalCount?: number
  onSaleCount?: number
}

export interface ProductPackageQueryModel {
  name?: string
  searchKey?: string
  onlineSale?: boolean
  isOnlineSaleMicoSchool?: boolean
  isShowMicoSchool?: boolean
  productPackageProperties?: ProductPackagePropertyRef[]
}

export interface ProductPackageQueryParams {
  pageRequestModel: {
    pageSize: number
    pageIndex: number
    needTotal?: boolean
    skipCount?: number
  }
  sortModel?: Record<string, any>
  queryModel?: ProductPackageQueryModel
}

export interface ProductPackageCreateParams {
  name: string
  onlineSale: boolean
  isAllowEditWhenEnroll: boolean
  title: string
  images: string
  description: string
  isShowMicoSchool: boolean
  isOnlineSaleMicoSchool: boolean
  buyRule: Record<string, any>
  items: ProductPackageItemMutation[]
  subjectIds: number[]
  productPackageProperties: ProductPackagePropertyRef[]
}

export function getProductPackagePagedListApi(data: ProductPackageQueryParams) {
  return usePost<ProductPackagePagedResult>('/api/v1/product-packages/page', data)
}

export function getProductPackageStatisticsApi(data: ProductPackageQueryModel) {
  return usePost<ProductPackageStatistics>('/api/v1/product-packages/statistics', data)
}

export function createProductPackageApi(data: ProductPackageCreateParams) {
  return usePost<string>('/api/v1/product-packages/create', data)
}
